### PR TITLE
HTTP server implementation

### PR DESCRIPTION
**Pull request by gustavosinbandera1**
this is the new implementation for Http server with dev as a base branch

### CODE REVIEW

File: components/NVS/src/Storage.cpp [hunk 1]
```diff
@@ -100,6 +100,18 @@ esp_err_t NVS::get_bool(const char* key, bool& value)
     return ESP_OK;
 }
 
+esp_err_t NVS::setString(std::string key, std::string data) 
+{
+    return nvs_set_str(m_handle, key.c_str(), data.c_str());
+}
```
Comment (jeandudey) [inline components/NVS/src/Storage.cpp]:
```suggestion
esp_err_t NVS::setString(const char* key, std::string data) 
{
    return nvs_set_str(m_handle, key, data.c_str());
}
```

Comment (jeandudey) [reply components/NVS/src/Storage.cpp]: `std::string` allocates memory in the **heap**, here we don't need that for the key.

File: components/NVS/src/Storage.cpp [hunk 2]
```diff
@@ -100,6 +100,18 @@ esp_err_t NVS::get_bool(const char* key, bool& value)
...
+
+esp_err_t NVS::getString(std::string key,  char* buffer, size_t* length)
+{
+
+    return  nvs_get_str(m_handle, key.c_str(), buffer, length);
```
Comment (jeandudey) [inline components/NVS/src/Storage.cpp]:
```suggestion
esp_err_t NVS::getString(const char* key,  std::string* data)
{
    esp_err_t err;
    std::size_t required_size;
    if ((err = nvs_get_str(m_handle, key.c_str(), nullptr, &required_size)) != ESP_OK) {
        return err;
    }
    data->clear();
    data->reserve(required_size);
    data->resize(required_size, '\0');
    return  nvs_get_str(m_handle, key, data->data(), &required_length);
```

Comment (jeandudey) [reply components/NVS/src/Storage.cpp]: For type safety, convert directly to a `std::string` and to maintain consistency. Can be done later if you want. I'm ok merging this PR as is.

File: main/defaults.h [hunk 3]
```diff
@@ -38,4 +38,10 @@
 #define WAP_MAXCONN 4                       /*!< AP Maximum connections */
 #define WAP_AUTHMODE WIFI_AUTH_WPA_WPA2_PSK /*!< AP Authentication mode */
 
+#define USER_NAME_KEY "username"                   /*default user_name for app to do request to server*/
```
Comment (jeandudey) [inline main/defaults.h]:
```suggestion
#define USER_NAME_KEY "username"                   /*!< default user_name for app to do request to server*/
```

Use documentation style for comments.

File: main/main.cpp [hunk 4]
```diff
@@ -18,22 +18,69 @@
 #include <freertos/task.h>
 
 #include <BLEPreferences.h>
```
Comment (jeandudey) [inline main/main.cpp]: Remove this

File: main/main.cpp [hunk 5]
```diff
@@ -18,22 +18,69 @@
...
+    } else {
+        err = app_nvs.getString(USER_PASSWORD_KEY, user_password, &pl);
+        if (err != ESP_OK) {
+            err = app_nvs.setString(USER_PASSWORD_KEY, DEFAULT_USER_PASSWORD);
+            if (err != ESP_OK) {
+                return ESP_FAIL;
+            }
+        }
+        ESP_LOGI(TAG, "\n");
+        ESP_LOGI(TAG, "Username: %s, Password: %s --> inside eeprom", &user_name[0], &user_password[0]);
```
Comment (luisan00) [inline main/main.cpp]:
```suggestion
        ESP_LOGI(TAG, "\nUsername: %s, Password: %s --> inside eeprom", &user_name[0], &user_password[0]);
```

**Pull request merged**